backend/app/services/knowledge/task_knowledge_base_service.py

```
@@ -147,67 +150,54 @@ def _is_kb_bound_to_user_group_chat(
+        from app.models.task_kb_binding import TaskKnowledgeBaseBinding
+
+        # Use EXISTS with the bindings table for efficient indexed query
+        # Check if there's any binding where:
+        # 1. The KB matches the given kb_id
+        # 2. The task is either owned by the user OR the user is an approved member
+        # Subquery: tasks owned by user
+        owned_task_subquery = (
+            select(TaskResource.id)
+            .where(
                 TaskResource.user_id == user_id,
+                TaskResource.is_active == True,
```

Ruff (0.15.5)

[error] Avoid equality comparisons to `True`; use `TaskResource.is_active:` for truth checks

Replace with `TaskResource.is_active`

(E712)